openupgrade_scripts/scripts/loyalty/16.0.1.0/post-migration.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from odoo.tools.translate import _
+
+
+def convert_loyalty_program_rewards(env):
+    openupgrade.m2o_to_x2m(
+        env.cr, env["loyalty.program"], "loyalty_program", "reward_ids", "reward_id"
+    )
+
+
+def convert_loyalty_program_rules(env):
+    openupgrade.m2o_to_x2m(
+        env.cr, env["loyalty.program"], "loyalty_program", "rule_ids", "rule_id"
+    )
+
+
+def compute_portal_point_name(env):
+    programs = env["loyalty.program"].search([])
```

Comment:
This migration is going to be quite big. Document well the why's behing these methods

Reply:
I think you can do this with SQL.

Reply:
I'm going to tune as much as possible to do as much as I can in SQL.